pkg/storage/tsdb/cacheable_index.go

```diff
@@ -54,10 +55,15 @@ func (i *CacheableIndex) GetChunkRefs(ctx context.Context, userID string, from,
 		if shard != nil {
 			stringfiedShard = shard.String()
 		}
-		if matcher == nil {
-			return userID + sep + from.String() + sep + through.String() + stringfiedShard + sep
+
+		keyMembers := []string{
+			userID, from.String(), through.String(), stringfiedShard,
```

Comment:
should we declare this above `if shard != nil` and then only append `stringifiedShard` to `keyMembers` inside that same if statement?

Reply:
hmm this was a good change for the matcher but I think it won't necessarily be for the shard because we always want the shard, even if it is nil/empty string.

That said, although the number of lines/logic/readability will be the same, I don't have a strong preference for defining the `keyMembers` before or after it. So, double checking, do you prefer if I do the keyMembers assignment before the shard conditional clause?